src/Gui/MainWindow.h

```diff
@@ -77,6 +77,13 @@ class GuiExport MainWindow : public QMainWindow
     Q_OBJECT
 
 public:
+    enum ConfirmSaveResult {

```

Comment:
I think you can make this an `enum class` with no other code changes (since you already specify the full scope when using it).

Reply:
I tried doing it but there were some complications because this is also a Qt slot.